kythe/typescript/indexer.ts

```diff
@@ -638,11 +638,21 @@ class Visitor {
 
     if (decl.typeParameters) this.visitTypeParameters(decl.typeParameters);
     this.visitType(decl.type);
-    // TODO: in principle right here we emit an "aliases" edge.
-    // However, it's complicated by the fact that some types don't have
-    // specific names to alias, e.g.
-    //   type foo = number|string;
-    // Just punt for now.
+
+    // Emit an "aliases" edge if the aliased type is a singleton and named in
```

Comment:
I think the meaning of "singleton" here must be unusual: Does it matter that the type has only one element? Or is this a TypeScript-ism for a type that is not a union?

Reply:
> Does it matter that the type has only one element?

Yeah, this is what I mean to say. Is there a better name for this?

Reply:
I don't know if there is a term of art for this. Perhaps "If the aliased type is a user-defined, named type that is not a union"?  User-defined rules out builtins.

"Singleton" usually refers to the members of the type, rather than the structure—but I have noticed that TS uses some other type-related terms in a slightly unusual way too, so I wasn't sure if this was another case of that.